src/metabase/actions/execution.clj

```diff
@@ -192,31 +203,47 @@
       (execute-implicit-action! action request-parameters)
       (:query :http)
       (execute-custom-action! action request-parameters)
-      (throw (ex-info (tru "Unknown action type {0}." (name (:type action))) action)))))
+      (throw (ex-info (tru "Unknown action type {0}." (name (:type action :unknown))) action)))))
+
+(defn- execute-table-action! [action-kw table-id request-parameters]
+  ;; removing nils for update is a tmp hack to ignore inputs that have not been interacted with from the FE
+  ;; we should have the FE omit these keys longer term
+  (let [request-parameters' (case action-kw
+                              :table.row/update (u/remove-nils request-parameters)
+                              request-parameters)]
+    (actions/perform-action!
+     action-kw
+     {:table-id table-id :row request-parameters'}
+     :policy :data-editing)))
 
 (mu/defn execute-dashcard!
   "Execute the given action in the dashboard/dashcard context with the given parameters
    of shape `{<parameter-id> <value>}."
   [dashboard-id       :- ::lib.schema.id/dashboard
    dashcard-id        :- ::lib.schema.id/dashcard
    request-parameters :- [:maybe [:map-of :string :any]]]
-  (let [dashcard (api/check-404 (t2/select-one :model/DashboardCard
-                                               :id dashcard-id
-                                               :dashboard_id dashboard-id))
-        action (api/check-404 (action/select-action :id (:action_id dashcard)))]
-    (analytics/track-event! :snowplow/action
-                            {:event     :action-executed
-                             :source    :dashboard
-                             :type      (:type action)
-                             :action_id (:id action)})
-    (execute-action! action request-parameters)))
+  (let [dashcard     (api/check-404 (t2/select-one :model/DashboardCard :id dashcard-id :dashboard_id dashboard-id))
+        action-id    (:action_id dashcard)
+        table-action (-> dashcard :visualization_settings :table_action)]
+    (api/check-404 (or action-id table-action))
+    (if table-action
+      (let [{:keys [kind table_id]} table-action]
+        ;; avoiding snowplow for now, to avoid adding new actions into their schema
+        (execute-table-action! (keyword kind) table_id request-parameters))
+      (let [action (api/check-404 (action/select-action :id action-id :archived false))]
+        (analytics/track-event! :snowplow/action
+                                {:event     :action-executed
+                                 :source    :dashboard
+                                 :type      (:type action)
+                                 :action_id (:id action)})
+        (execute-action! action request-parameters)))))
```

Comment:
We should revert this, not adding support for table actions in dashboard buttons (yet)